src/main/java/org/sonarqube/gradle/AndroidUtils.java

```diff
@@ -101,6 +108,30 @@ private static void configureForAndroid(Project project, BaseVariant variant, Ma
     }
   }
 
+  private static void populateSonarQubeAndroidProperties(AndroidVariantAndExtension android, Map<String, Object> properties) {
+    properties.put(AndroidProperties.ANDROID_DETECTED, true);
+
+    if(!isMinSdkSupported()) {
+      return;
+    }
+
+    if (android.isVariantProvidedByUser()) {
+      ApiVersion minSdkVersion = android.getVariant().getMergedFlavor().getMinSdkVersion();
+      if (minSdkVersion != null) {
+        properties.put(AndroidProperties.MIN_SDK_VERSION_MIN, minSdkVersion.getApiLevel());
+        properties.put(AndroidProperties.MIN_SDK_VERSION_MAX, minSdkVersion.getApiLevel());
+      }
+    } else {
+      Set<Integer> minSdks = android.getExtension().getProductFlavors().stream().map(ProductFlavor::getMinSdk).collect(Collectors.toSet());
+      minSdks.add(android.getExtension().getDefaultConfig().getMinSdk());
+      minSdks.remove(null);
```

Comment:
Can we instead don't add the null entry instead of deleting it afterward

Reply:
We could. This would require us to check both in line 125 and in line 126 for null or write our own implementation of Set that ignores nulls.

Reply:
I think that immutability is preferred when we can
Other solution 
```java
Stream.concat ( android.getExtension().getProductFlavors().stream(), Stream.of (ndroid.getExtension().getDefaultConfig().getMinSdk()))
.filter(Objects::nonNull)
.collect(Collectors.toSet())
```

I recon this does not make it necessarily clearer

Reply:
Fixed